src/hooks/useEntitlement.ts

```diff
@@ -0,0 +1,51 @@
+import { useQueries } from 'react-query';
+import { useMemo } from 'react';
+
+import type { GetEntitlementsResponse } from '../../types/checkout';
+import type { MediaOffer } from '../../types/media';
+
+import { useConfigStore } from '#src/stores/ConfigStore';
+import { useAccountStore } from '#src/stores/AccountStore';
+import { getEntitlements } from '#src/services/checkout.service';
+
+export type UseEntitlementResult = {
+  isEntitled: boolean;
+  isLoading: boolean;
+  error: unknown | null;
+};
+
+export type UseEntitlement = (mediaOffers?: MediaOffer[], enabled?: boolean) => UseEntitlementResult;
+
+type QueryResult = {
+  responseData?: GetEntitlementsResponse;
+};
+
+const useEntitlement: UseEntitlement = (mediaOffers = [], enabled = true) => {
+  const sandbox = useConfigStore(({ config }) => config?.cleengSandbox);
+  const jwt = useAccountStore(({ auth }) => auth?.jwt);
+
+  const entitlementQueries = useQueries(
```

Comment:
Won't this result in caching the entitlement results even after potentially the user has just purchased an entitlement? I feel like there needs to be some invalidation logic somewhere?

Reply:
Is the entitlement call that slow that the query caching is valuable?  Or are there other benefits that I'm not thinking of?

Reply:
I guess we don't need the caching mechanism for this indeed. @royschut we better just call the entitlement service directly without `useQueries` in this case.

Reply:
The call is quite fast actually, so sure, I'll take `useQueries()` out.

Reply:
So as discussed below, we continue with `useQueries` for now and try out a proper invalidation first.